src/charm.py

```diff
@@ -228,6 +231,11 @@ def _check_precondition(self) -> None:
             raise InvalidIngressError(
                 "nginx-ingress-integrator cannot establish more than one relation at a time"
             )
+        if ingress_relation is not None and len(hostnames) > 1:
```

Comment:
Why two hostnames and not one?

Reply:
we want to make sure that the number of hostnames is 1. if there are any more than 1 hostname (and we are using the ingress relation) then we should raise an error, so that the charm is blocked.

Reply:
Gotcha, I was reading it wrong